### PR TITLE
drivers: spi: Add Intel PCH spi penwell driver

### PR DESCRIPTION
**Pull request by rbabub**
Submitting this PR for review of Intel pch penwell SPI driver.

This PR includes commits for
- The driver code
- Dts node for SPI instances
- SPI loopback test conf and overlay file

Signed-off-by: Ramesh Babu B <ramesh.babu.b@intel.com>

### CODE REVIEW

File: tests/drivers/spi/spi_loopback/boards/rpl_crb.conf [hunk 1]
```diff
@@ -0,0 +1,6 @@
+CONFIG_SPI=y
+CONFIG_SPI_PW=y
```
Comment (MaureenHelm) [inline tests/drivers/spi/spi_loopback/boards/rpl_crb.conf:2]: This line isn't necessary

File: tests/drivers/spi/spi_loopback/boards/rpl_crb.conf [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+CONFIG_SPI_ASYNC=n
+CONFIG_GPIO=y
+CONFIG_GPIO_INTEL=y
```
Comment (MaureenHelm) [inline tests/drivers/spi/spi_loopback/boards/rpl_crb.conf:5]: This line isn't necessary

**Review by MaureenHelm — COMMENTED** [review]
A few nits, but otherwise looks great

File: drivers/spi/spi_pw.c [hunk 3]
```diff
@@ -0,0 +1,895 @@
...
+	ctrlr1 = spi_pw_reg_read(dev, PW_SPI_REG_CTRLR1);
+	if (rx_mask) {
+		ctrlr1 |= PW_SPI_INTR_BITS;
+	} else {
+		ctrlr1 |= PW_SPI_INTR_BITS;
+		ctrlr1 &= ~(PW_SPI_INTR_MASK_RX);
+	}
+	spi_pw_reg_write(dev, PW_SPI_REG_CTRLR1, ctrlr1);
+}
+#endif
```
[the remark this answers is not in the document]
Comment (rbabub) [reply drivers/spi/spi_pw.c]: > should this move down below `spi_pw_intr_disable`?

resolved.

File: drivers/spi/Kconfig.pw [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+if SPI_PW
+
+config SPI_PW_FIFO_DEPTH
+	int "RX and TX FIFO Depth"
+	default 64
+	help
+	  Corresponds to the SSP_TX_FIFO_DEPTH and
+	  SSP_RX_FIFO_DEPTH of the Penwell Synchronous
+	  Serial Interface. Depth ranges from 2-64.
```
[the remark this answers is not in the document]
Comment (rbabub) [reply drivers/spi/Kconfig.pw]: > Move this to a devicetree property

resolved.

File: tests/drivers/spi/spi_loopback/boards/rpl_crb.conf (hunk 1 above)
Comment (rbabub) [reply tests/drivers/spi/spi_loopback/boards/rpl_crb.conf]: > This line isn't necessary

resolved.

File: tests/drivers/spi/spi_loopback/boards/rpl_crb.conf (hunk 2 above)
Comment (rbabub) [reply tests/drivers/spi/spi_loopback/boards/rpl_crb.conf]: > This line isn't necessary

resolved.

File: drivers/spi/spi_pw.c [hunk 5]
```diff
@@ -0,0 +1,895 @@
...
+
+	spi_context_unlock_unconditionally(&spi->ctx);
+
+	LOG_DBG("SPI pw init success");
+
+	return 0;
+}
+
+#define INIT_PCIE0(n)
+#define INIT_PCIE1(n) DEVICE_PCIE_INST_INIT(n, pcie),
```
[the remark this answers is not in the document]
Comment (rbabub) [reply drivers/spi/spi_pw.c:814]: > this doesn't appear to be used

INIT_PCIE(n) macro is called from the driver init.

**Review by jhedberg — APPROVED** [review]

**Review by finikorg — APPROVED** [review]

**Pull request merged**